### PR TITLE
Ignore slashes after the protocol for special URLs

### LINKED ISSUE #205: URLs with no hostname not parsed properly in v1.5.0
Hi there, I'm using this library to parse URLs with a custom protocol like so:

```js
my-custom-protocol:///path-to/image.jpg
```

The behaviour has changed in v1.5.0 so that part of the pathname is being treated as the host. I believe this is a bug and goes against the browser implementation of `URL`, which produces a similar output similar to v1.4.7 with the `hostname` unset. I think it's due to the change in regex match [here](https://github.com/unshiftio/url-parse/blob/master/index.js#L6).

Output in v1.4.7:

```js
{
  slashes: true,
  protocol: 'my-custom-protocol:',
  hash: '',
  query: '',
  pathname: '/path-to/image.jpg',
  auth: '',
  host: '',
  port: '',
  hostname: '',
  password: '',
  username: '',
  origin: 'null',
  href: 'my-custom-protocol:///path-to/image.jpg'
}
```

Output in v1.5.0:

```js
{
  slashes: true,
  protocol: 'my-custom-protocol:',
  hash: '',
  query: '',
  pathname: '/image.jpg',
  auth: '',
  host: 'path-to',
  port: '',
  hostname: 'path-to',
  password: '',
  username: '',
  origin: 'my-custom-protocol://path-to',
  href: 'my-custom-protocol://path-to/image.jpg'
}
```

Let me know if you need any more details. Thanks!

### LINKED ISSUE #206: Security issues Hostname spoofing & Open Redirect
@lpinca @3rd-Eden I have reported a security issue in huntr
https://www.huntr.dev/bounties/1625557993985-unshiftio/url-parse/ 
There are 2 attack scenarios possible for **Open Redirect** and **Hostname Spoofing(Please take a look at the last comment)**

Please validate and let us know your opinion on this. Thank you.

### PR DESCRIPTION
**Pull request by lpinca**
Fixes #205
Fixes #206

### CODE REVIEW

**Comment by lpinca** [comment]
I have to add tests but it should work.

**Comment by 3rd-Eden** [comment]
LGTM when additional tests are added.

File: test/test.js [hunk 1]
```diff
@@ -292,11 +292,59 @@ describe('url-parse', function () {
...
+    parsed = parse(url);
+    assume(parsed.hostname).equals('');
+    assume(parsed.pathname).equals('/example.com');
+    assume(parsed.href).equals('foo:/example.com');
+
+    url = 'foo://example.com';
+    parsed = parse(url);
+    assume(parsed.hostname).equals('example.com');
+    assume(parsed.pathname).equals('/');
+    assume(parsed.href).equals('foo://example.com/');
```
Comment (lpinca) [inline test/test.js:339]: Behavior is not consistent with the Node.js WHATWG URL parser.

```
new URL('foo://example.com')
URL {
  href: 'foo://example.com',
  origin: 'null',
  protocol: 'foo:',
  username: '',
  password: '',
  host: 'example.com',
  hostname: 'example.com',
  port: '',
  pathname: '',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

```
> parse('foo://example.com')
{
  slashes: true,
  protocol: 'foo:',
  hash: '',
  query: '',
  pathname: '/',
  auth: '',
  host: 'example.com',
  port: '',
  hostname: 'example.com',
  password: '',
  username: '',
  origin: 'foo://example.com',
  href: 'foo://example.com/'
}
```

Our `pathname` is `'/'` instead of `''`.

Also note `origin` which should be `'null'` and not `'foo://example.com'`. It should be set only if the URL is special. Anyway this should be easy to fix in a different PR/commit.

File: test/test.js [hunk 2]
```diff
@@ -292,11 +292,59 @@ describe('url-parse', function () {
...
+
+    assume(parsed.hostname).equals('');
+    assume(parsed.pathname).equals('example.com');
+    assume(parsed.href).equals('foo:example.com');
+
+    url = 'foo:/example.com';
+    parsed = parse(url);
+    assume(parsed.hostname).equals('');
+    assume(parsed.pathname).equals('/example.com');
+    assume(parsed.href).equals('foo:/example.com');
```
Comment (lpinca) [inline test/test.js:333]: Behavior is not consistent with the Node.js WHATWG URL parser if backslashes are used:

```
new URL('foo:\\example.com')
URL {
  href: 'foo:\\example.com',
  origin: 'null',
  protocol: 'foo:',
  username: '',
  password: '',
  host: '',
  hostname: '',
  port: '',
  pathname: '\\example.com',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

```
parse('foo:\\example.com')
{
  slashes: false,
  protocol: 'foo:',
  hash: '',
  query: '',
  pathname: '/example.com',
  auth: '',
  host: '',
  port: '',
  hostname: '',
  password: '',
  username: '',
  origin: 'null',
  href: 'foo:/example.com'
}
```

**Comment by lpinca** [comment]
Besides https://github.com/unshiftio/url-parse/pull/208#discussion_r675783190 and https://github.com/unshiftio/url-parse/pull/208#discussion_r675788224 there are a lot of other inconsistencies with the Node.js WHATWG URL parser. I don't think it's possible to fix all of them unless we rewrite `url-parse` from scratches following the specification.

File: test/test.js (hunk 2 above)
Comment (lpinca) [reply test/test.js:358]: Fixed in eda1342 and e24fca3.

File: test/test.js (hunk 1 above)
Comment (lpinca) [reply test/test.js]: Fixed in https://github.com/unshiftio/url-parse/pull/208/commits/eda1342a352b980710e1947e7b0c617e53894310 and https://github.com/unshiftio/url-parse/pull/208/commits/e24fca31c61cc38f912c87225e28251d1b72599e.

**Comment by lpinca** [comment]
@3rd-Eden I'm merging this. Can you please cut a new release?

File: test/test.js (hunk 2 above)
Comment (lpinca) [reply test/test.js:358]: Fixed in https://github.com/unshiftio/url-parse/pull/209/commits/48bdd73271eceec58b094e92a737b5028d61879a.
